### PR TITLE
add user page

### PR DESCRIPTION
**Pull request by iSharovatov1**
# [newsSiteFront](https://github.com/iSharovatov1/newsSiteFront)
> Creating a user page (server + client)
### Code Details
* add user page
* ### Screenshots

### Feature tag selection
Has a `Creating a user page (server + client)`  Has it been added [to the tracker](https://trello.com/b/EICN697S/%D1%88%D0%B0%D1%80%D0%BE%D0%B2%D0%B0%D1%82%D0%BE%D0%B2%D0%B8%D0%B3%D0%BE%D1%80%D1%8C%D0%BD%D0%BE%D0%B2%D0%BE%D1%81%D1%82%D0%BD%D0%BE%D0%B9%D1%81%D0%B0%D0%B9%D1%82)?

### CODE REVIEW

File: src/store/reducers/userReducer.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  user: {},
+  news: [],
+  loading: false,
+  error: null,
+};
+
+function reducers(state = initialState, action = {}) {
+  switch (action.type) {
+    case ACTION_TYPES.USER_FETCH_REQUESTED:
+      return { ...state, loading: true, error: null };
```
Comment (tatiana-dunice) [inline src/store/reducers/userReducer.js:13]: you're using the same loading field for sevaral requests - can couse an issue

**Review by tatiana-dunice — CHANGES_REQUESTED** [review]

File: src/api/newsApi.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import instance from './instance';
 
 const getNews = async () => {
-  const { data } = await instance.get('/api/news');
+  const { data } = await instance.get('/news');
```
[the remark this answers is not in the document]
Comment (iSharovatov1) [reply src/api/newsApi.js:4]: I changed path in env

[the remark this answers is not in the document]
Comment (tatiana-dunice) [reply src/api/newsApi.js:4]: got it

**Review by tatiana-dunice — CHANGES_REQUESTED** [review]

**Review by tatiana-dunice — CHANGES_REQUESTED** [review]

File: src/api/instance.js [hunk 3]
```diff
@@ -2,7 +2,6 @@ import { create } from 'axios';
 
 const instance = create({
   baseURL: process.env.REACT_APP_BASE_URL,
```
Comment (tatiana-dunice) [inline src/api/instance.js:4]: why did you remove `withCredentials`?

File: src/components/NewsCard/NewsCard.jsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+import { CircularProgress } from '@mui/material';
+
+import styles from './NewsCard.module.scss';
+
+function NewsCard({
+  news,
+  onClick,
+  isLoading = false,
+}) {
+  const defaultImg = '/assets/images/defaultImg.png';
```
Comment (tatiana-dunice) [inline src/components/NewsCard/NewsCard.jsx:13]: use it in place or move outside the component - and change name - it's a constant

File: src/components/NewsCard/NewsCard.jsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  return (
+    <div>
+      { isLoading
```
Comment (tatiana-dunice) [inline src/components/NewsCard/NewsCard.jsx:16]: You can avoid this internal conditions by moving it outside

File: src/components/NewsCard/NewsCard.module.scss [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        display: flex;
+        max-width: 500px;
+        .tag {
+          border: 1px solid lightgray;
+          border-radius: 4px;
+          padding: 5px;
+          margin: 0 5px;
+        }
+      }
+      .autor {
```
Comment (tatiana-dunice) [inline src/components/NewsCard/NewsCard.module.scss:57]: redundant style - remove it

**Review by tatiana-dunice — CHANGES_REQUESTED** [review]

File: src/components/NewsCard/NewsCard.jsx [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+              </div>
+            )}
+        </div>
+      </div>
+    </div>
+  );
+}
+
+NewsCard.defaultProps = {
+  onClick: () => null,
```
Comment (tatiana-dunice) [inline src/components/NewsCard/NewsCard.jsx:37]: `() => null,` -> `() => {},`

File: src/components/UserProfile/UserProfile.module.scss [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+    .feildContainer {
+      width: 50%;
+      display: flex;
+      justify-content: space-between;
+      font-size: 20px;
+      font-weight: 500;
+      padding: 10px 80px;
+    }
+  }
+}
```
Comment (tatiana-dunice) [inline src/components/UserProfile/UserProfile.module.scss:32]: New line is required

**Review by tatiana-dunice — CHANGES_REQUESTED** [review]

**Review by tatiana-dunice — APPROVED** [review]

**Pull request merged**